packages/esm-patient-orders-app/src/components/test-order.component.tsx

```diff
@@ -47,63 +47,74 @@ const TestOrder: React.FC<TestOrderProps> = ({ testOrder }) => {
   );
 
   const testResultObs = useMemo(() => {
-    if (!encounter || !concept) return null;
-    return encounter.obs?.find((obs) => obs.concept.uuid === concept.uuid);
-  }, [concept, encounter]);
+    if (!encounter) return null;
+    return encounter.obs?.filter((obs) => obs.order.uuid === testOrder.uuid);
+  }, [encounter, testOrder.uuid]);
+
+  const obsUuids = useMemo(
+    () =>
+      Array.isArray(testResultObs) && testResultObs.length > 0
+        ? testResultObs.map((o) => o.concept?.uuid).filter(Boolean)
+        : [],
+    [testResultObs],
+  );
```

Comment:
It seems to me that rather than _chaining_ `useMemo()`s like this it's better to write:

```ts
const [testResultObs, obsUuids] = useMemo(() => {
  if (!encounter) return [[], []];
  const testObs = encounter.obs?.filter((obs) -> obs.order?.uuid === testOrder.uuid) ?? [];
  return [testObs, testObs.map((obs) => obs.concept?.uuid).filter(Boolean)];
}, [encounter, testOrder.uuid]);
```